File: GGAndroid_2.0/GGDemo/src/main/java/io/ganguo/demo/view/activity/MainActivity.java
package io.ganguo.demo.view.activity;

import io.ganguo.demo.view.activity.base.GGHFRVModelActivity;
import io.ganguo.demo.view.activity.base.GGHFSRVModelActivity;
import io.ganguo.demo.viewmodel.activity.MainVModel;
import io.ganguo.vmodel.ViewModelActivity;
import io.ganguo.viewmodel.databinding.IncludeHfRecyclerBinding;

/**
 * Main Activity
 */
public class MainActivity extends GGHFRVModelActivity<IncludeHfRecyclerBinding, MainVModel> {

    @Override
    public MainVModel createViewModel() {
        return new MainVModel();
    }

    @Override
    public void onViewAttached(MainVModel viewModel) {
        super.onViewAttached(viewModel);
    }

}

File: GGAndroid_3.0/main/app-mvvm-v1-sample/src/main/java/io/ganguo/demo/view/activity/MainActivity.kt
package io.ganguo.demo.view.activity

import io.ganguo.demo.R
import io.ganguo.demo.view.activity.base.GGHFRVModelActivity
import io.ganguo.demo.viewmodel.activity.MainVModel
import io.ganguo.log.core.Logger
import io.ganguo.utils.util.Apps
import io.ganguo.viewmodel.databinding.IncludeHfRecyclerBinding

/**
 * Main Activity
 */
class MainActivity : GGHFRVModelActivity<IncludeHfRecyclerBinding, MainVModel>() {

    override fun createViewModel(): MainVModel {
        return MainVModel()
    }


    override fun getBackDrawableRes(): Int {
        return 0
    }

    override fun onBackPressed() {
        Apps.exitByDoublePressed(this)
    }


}
